fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/QueryTransformer.java

```diff
@@ -288,13 +289,26 @@ private OptExprBuilder limit(OptExprBuilder subOpt, LimitElement limit) {
         return subOpt.withNewRoot(limitOperator);
     }
 
-    private OptExprBuilder aggregate(OptExprBuilder subOpt, List<Expr> groupByExpressions,
-                                     List<FunctionCallExpr> aggregates,
+    private OptExprBuilder aggregate(OptExprBuilder subOpt,
+                                     List<Expr> groupByExpressions, List<FunctionCallExpr> aggregates,
                                      List<List<Expr>> groupingSetsList, List<Expr> groupingFunctionCallExprs) {
         if (aggregates.size() == 0 && groupByExpressions.size() == 0) {
             return subOpt;
         }
 
+        // handle aggregate function use grouping set columns, use clone column replace original column
+        // e.g:
+        // before: select sum(a) from xx group by rollup(a);
+        // after: select sum(clone(a)) from xx group by rollup(a);
+        if (groupingSetsList != null) {
+            for (Expr groupBy : groupByExpressions) {
+                aggregates.forEach(f -> f.getParams().exprs()
+                        .replaceAll(root -> replaceExprBottomUp(root, groupBy, new CloneExpr(groupBy))));
```

Comment:
Could use visitor like NotFullGroupByRewriter

Reply:
It's different, `NotFullGroupByRewriter` only reslove `SlotExpr`, we need reslove function. Like:
```
select v1,sum(id + 1) from t1 group by rollup(v1, id + 1) order by v1;
```